### ISSUE TITLE
Rule proposal: no-one-time-vars

### ISSUE THREAD

**Issue by Bas950** [typescript-eslint/typescript-eslint#4118]
### Description

Errors when a var is only used once.

### Fail

```ts
var once = 'used';
console.log(once);
```

### Pass

```ts
var once = 'used';
console.log(once);
// Do other stuff
console.log(once);
```

**Comment by bradzacher** [comment]
As per the issue form you filled out

> Ensure that the rule you want to propose is related to types

There is nothing here specific to typescript or that requires type information. Thus it is not a good fit to exist within this project.

**Comment by Bas950** [comment]
> As per the issue form you filled out
> 
> > Ensure that the rule you want to propose is related to types
> 
> There is nothing here specific to typescript or that requires type information. Thus it is not a good fit to exist within this project.

Where should I suggest it in your opinion?

**Comment by bradzacher** [comment]
eslint core maybe - though this is *very* stylistic, so I don't think they'd accept it as they are not accepting any new stylistic lint rules.

eslint-plugin-unicorn maybe?
